src/mc-fle2-payload-uev.c

```diff
@@ -172,6 +172,52 @@ mc_FLE2UnindexedEncryptedValue_decrypt (_mongocrypt_crypto_t *crypto,
    return &uev->plaintext;
 }
 
+bool
+mc_FLE2UnindexedEncryptedValue_encrypt (_mongocrypt_crypto_t *crypto,
+                                        const _mongocrypt_buffer_t *key_uuid,
+                                        bson_type_t original_bson_type,
+                                        const _mongocrypt_buffer_t *plaintext,
+                                        const _mongocrypt_buffer_t *key,
+                                        _mongocrypt_buffer_t *out,
+                                        mongocrypt_status_t *status)
+{
+   _mongocrypt_buffer_t iv = {0};
+   bool res = false;
+
+   _mongocrypt_buffer_resize (&iv, MONGOCRYPT_IV_LEN);
+   if (!_mongocrypt_random (crypto, &iv, MONGOCRYPT_IV_LEN, status)) {
+      goto fail;
```

Comment:
This `goto` will attempt to cleanup the `AD` variable before control has passed over its zero-initialization below.

Reply:
Good catch! Moved the declaration of `AD` to the start.